doc/devel/documenting_mpl.rst

```diff
@@ -91,270 +87,108 @@ statement, such as::
-documentation of every function that takes a ``**kwargs``.  The requirements
-are:
-
-1. single point of configuration so changes to the properties don't
-   require multiple docstring edits.
-
-2. as automated as possible so that as properties change, the docs
-   are updated automagically.
-
-The function `matplotlib.artist.kwdoc` and the decorator
-`matplotlib.docstring.dedent_interpd` facilitate this.  They combine python
-string interpolation in the docstring with the Matplotlib artist introspection
-facility that underlies ``setp`` and ``getp``.  The ``kwdoc`` function gives
-the list of properties as a docstring. In order to use this in another
-docstring, first update the ``matplotlib.docstring.interpd`` object, as seen in
-this example from `matplotlib.lines`:
-
-.. code-block:: python
-
-  # in lines.py
-  docstring.interpd.update(Line2D=artist.kwdoc(Line2D))
-
-Then in any function accepting `~.Line2D` pass-through ``kwargs``, e.g.,
-`matplotlib.axes.Axes.plot`:
-
-.. code-block:: python
-
-  # in axes.py
-  @docstring.dedent_interpd
-  def plot(self, *args, **kwargs):
-      """
-      Some stuff omitted
-
-      The kwargs are Line2D properties:
-      %(Line2D)s
-
-      kwargs scalex and scaley, if defined, are passed on
-      to autoscale_view to determine whether the x and y axes are
-      autoscaled; default True.  See Axes.autoscale_view for more
-      information
-      """
-      pass
-
-Note there is a problem for `~matplotlib.artist.Artist` ``__init__`` methods,
-e.g., `matplotlib.patches.Patch.__init__`, which supports ``Patch`` ``kwargs``,
-since the artist inspector cannot work until the class is fully defined and
-we can't modify the ``Patch.__init__.__doc__`` docstring outside the class
-definition.  There are some some manual hacks in this case, violating the
-"single entry point" requirement above -- see the ``docstring.interpd.update``
-calls in `matplotlib.patches`.
-
 .. _formatting-mpl-docs:
 
 Formatting
 ==========
 
-The Sphinx website contains plenty of documentation_ concerning ReST markup and
-working with Sphinx in general. Here are a few additional things to keep in mind:
-
-* Please familiarize yourself with the Sphinx directives for `inline
-  markup`_. Matplotlib's documentation makes heavy use of cross-referencing and
-  other semantic markup. For example, when referring to external files, use the
-  ``:file:`` directive.
-
-* Function arguments and keywords should be referred to using the
-  ``*emphasis*`` role. This will keep Matplotlib's documentation consistent
-  with Python's documentation:
-
-  .. code-block:: rst
-
-     Here is a description of *argument*
-
-  Please do not use the ```default role```:
-
-  .. code-block:: rst
-
-     Please do not describe `argument` like this.
-
-  nor the ````literal```` role:
-
-  .. code-block:: rst
-
-     Please do not describe ``argument`` like this.
-
-* Mathematical expressions can be rendered as png images in html, and in the
```

Comment:
I've removed the maths, ipython, and footnotes bits, as they are covered in Sphinx documentation, and I don't think they're common enough to need a specific section in the Matplotlib docs.